src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -709,24 +731,22 @@ private class NativeMethods
             internal static extern ErrorCode rd_kafka_position(
                     IntPtr rk, IntPtr partitions);
 
+            // note: producev signature is rd_kafka_producev(rk, ...)
+            // __arglist is an undocumented keyword which deal with it
+            // but doesn't work on standardlib (and undocumented so don't use)
+            // stay simple with one binding for now, but it will be worth benchmarking
+            // the overload with no timestamp, opaque, partition, etc
             [DllImport(DllName, CallingConvention = CallingConvention.Cdecl)]
-            internal static extern IntPtr rd_kafka_produce(
-                    IntPtr rkt,
```

Comment:
actually, it would be possible to do a lookup into a dictionary of pre-serialised byte arrays instead of one of topic handles (though, come to think of it, that is not especially cheap with the string key, so we won't). anyway, my point is no need for producev to ever accept topic handles, please ignore my above comment.